### PR TITLE
Feature/farm 48 landing page

### PR DESCRIPTION
**Pull request by isaacmartinezimv**
[[FARM-48]-Landing Page
](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-48)
#### 📲 What

A landing page for the app

#### 🤔 Why
		
Because we need a introduction screen for the new users
		
#### 🛠 How
		
Applying the new code Style for Flutter

#### 👀 See
LandingPageGif3
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [x] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [x] Showcase video / automation test
- [ ] Passing all tests
- [x] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

Just Notice that the "real Landing page " will be in a full screen page, so now it's in playground and can confuse a bit, but it's been tested in a full screen page.

### CODE REVIEW

File: lib/ui/LandingPage.dart [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  String footerText;
+
+  String headerImage;
+  String subtitleImage;
+
+  LandingPageViewModel({this.detailText,
+    this.actionText,
+    this.headerImage,
+    this.subtitleImage,
+    this.footerText});
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Missing comma
```suggestion
    this.footerText,});
```

File: lib/ui/LandingPage.dart [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+class LandingPageStyle {
+  final TextStyle detailTextStyle;
+  final TextStyle footerTextStyle;
+
+  final EdgeInsets headerEdgePadding;
+  final EdgeInsets detailTextEdgePadding;
+  final EdgeInsets actionEdgePadding;
+  final EdgeInsets footerTextEdgePadding;
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Do we really need this on the style? I think we won't change the paddings of the text. I prefer to have it as a private const.

What do you think @LeeMatthewHiggins

File: lib/ui/LandingPage.dart [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+  final double subtitleLineSpace;
+  final double detailTextLineSpace;
+  final double actionLineSpace;
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Also these LineSpaces

File: lib/ui/LandingPage.dart [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    LandingPageViewModel viewModel,
+    LandingPageStyle style = _defaultStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SingleChildScrollView(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: why we need a scrollview for the landing page? Should be static page

File: lib/ui/LandingPage.dart [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            ),
+            _buildFooter(context),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Row _buildFooter(BuildContext context) {
+    return Row(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: I think that you can simplify it with a `Container` with a `width: double.infinity` instead of using a `Row` and then adding a `Expanded` widget

File: lib/ui/LandingPage.dart [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  const LandingPage({Key key,
+    LandingPageViewModel viewModel,
+    LandingPageStyle style = _defaultStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Scaffold is needed?? Don't think so

File: lib/ui/LandingPage.dart [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+      body: SingleChildScrollView(
+        child: Column(
+          children: <Widget>[
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Here I see two main containers the footer and the top. The bottom container that contains the button and the "language selector" and the other container that contains the header, subtitle and the detail text so you can use a `Column` with a `Expanded` and `flex: ` value  to avoid adding a lot of `Padding` on the child widgets

Screenshot 2019-07-16 at 12 46 31

File: lib/ui/LandingPage.dart [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+              viewModel: RoundedButtonViewModel(title: _viewModel.actionText),
+              style: RoundedButtonStyle.largeRoundedButtonStyle(),
+            ),
+          );
+  }
+
+  Padding _buildDetailText() {
+    return Padding(
+            padding: _style.detailTextEdgePadding,
+            child: Row(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Same here, you can achieve this with a `Container` with a `width: double.infinity`

File: lib/ui/LandingPage.dart [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+  }
+}
+
+Future _onMenuPressed(BuildContext context) async {
+  showModalBottomSheet(
+      context: context,
+      builder: (widgetBuilder) =>
+          ActionSheet(
+              viewModel: MockActionSheetViewModel.buildWithCheckBox(),
+              style: ActionSheetStyle.defaultStyle()));
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]:
```suggestion
              style: ActionSheetStyle.defaultStyle(),),);
```

File: lib/ui/common/CircularPainter.dart [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class CircularPainter extends CustomPainter{
```
Comment (davidfranquet) [inline lib/ui/common/CircularPainter.dart]: This class is part of this PR? 🤔 🤔 🤔 🤔 🤔

File: lib/ui/privacy_policies_screen.dart [hunk 11]
```diff
@@ -1,7 +1,4 @@
-
-
 import 'dart:convert';
-
```
Comment (davidfranquet) [inline lib/ui/privacy_policies_screen.dart]: I know that the CR are wrong. But this class is not part of this PR

File: pubspec.yaml [hunk 12]
```diff
@@ -87,8 +87,12 @@ flutter:
     - assets/icons/detail_icon_cost.png
     - assets/icons/radio_button_default.png
     - assets/icons/radio_button_active.png
-
-
+    - assets/raw/1.5x/
+    - assets/raw/2.0x/
+    - assets/raw/3.0x/
+    - assets/raw/4.0x/
```
Comment (davidfranquet) [inline pubspec.yaml]: we need this? or only the main folder? `assets/raw`

**Review by davidfranquet — CHANGES_REQUESTED** [review]

File: lib/ui/LandingPage.dart [hunk 13]
```diff
@@ -124,117 +127,99 @@ class LandingPage extends StatelessWidget {
...
-              height: _style.detailTextLineSpace
-            ),
-            _buildAction(context),
-            SizedBox(
-              height: _style.actionLineSpace
-            ),
-            _buildFooter(context),
-          ],
-        ),
+    return SingleChildScrollView(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Move this scrollview as a child of the `PlaygroundWidget` and add this widget as a child of the `SingleChildScrollView`

File: lib/ui/playground/playground_button_present_modal.dart [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  Widget build(BuildContext context) {
+    return Center(
+      child: RaisedButton(
+        child: Text("Press Me"),
+        onPressed: () => _onClick(context),
+      ),
+    );
+  }
+
+  _onClick(BuildContext context) {
```
Comment (LeeMatthewHiggins) [inline lib/ui/playground/playground_button_present_modal.dart]: we never click on mobile!!!!   :).

File: lib/ui/mockData/MockLandingPageViewModel.dart [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import 'package:farmsmart_flutter/ui/LandingPage.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static final String detailText = Intl.message("Kuwezesha wakenya elimu inayohitajika kuwa wakulima wenye mafanikio.");
```
Comment (LeeMatthewHiggins) [inline lib/ui/mockData/MockLandingPageViewModel.dart]: these are not final. if we change the language this string will not change!
Instead in the widget do Intl.message(_Strings.detailText) and in the _String class just have the english string.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]
check the translations comment.
plus @DavidFD005 davids comments. 
Looks good

File: lib/ui/LandingPage.dart [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+const LandingPageStyle _defaultStyle = const _DefaultStyle();
+
+class LandingPage extends StatelessWidget {
+  final LandingPageViewModel _viewModel;
+  final LandingPageStyle _style;
+
+  const LandingPage(
+      {Key key,
+      LandingPageViewModel viewModel,
+      LandingPageStyle style = _defaultStyle})
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]:
```suggestion
      LandingPageStyle style = _defaultStyle,})
```

File: lib/ui/LandingPage.dart [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: _Constants.generalPadding,
+      child: Column(
+        children: <Widget>[
+          Flexible(
+            flex: _Constants.secondaryPriority,
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]:
```suggestion
            flex: _Constants.imageFlexValue,
```

File: lib/ui/LandingPage.dart [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+                    ),
+                  ],
+                ),
+                Column(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: <Widget>[
+                    RoundedButton(
+                      viewModel: RoundedButtonViewModel(
+                          title: Intl.message(_viewModel.actionText),
+                          onTap: () {}),
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]:
```suggestion
                          onTap: () {},),
```

File: pubspec.yaml [hunk 19]
```diff
@@ -88,8 +88,9 @@ flutter:
     - assets/icons/detail_icon_cost.png
     - assets/icons/radio_button_default.png
     - assets/icons/radio_button_active.png
-
-
+    - assets/raw/
+    - assets/raw/illustration_welcome.png
```
Comment (davidfranquet) [inline pubspec.yaml]: just add `assets/raw/`

File: lib/ui/mockData/MockLandingPageViewModel.dart [hunk 20]
```diff
@@ -0,0 +1,20 @@
+import 'package:farmsmart_flutter/ui/LandingPage.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static String detailText = ("A network and knowledge source for farmers in Kenya");
```
Comment (LeeMatthewHiggins) [inline lib/ui/mockData/MockLandingPageViewModel.dart]: why the () ?

File: lib/ui/LandingPage.dart [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+                      textAlign: TextAlign.center,
+                    ),
+                  ],
+                ),
+                Column(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: <Widget>[
+                    RoundedButton(
+                      viewModel: RoundedButtonViewModel(
+                          title: Intl.message(_viewModel.actionText),
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart]: I think we should have the view model handle this........ the ViewModel should give the exact t string to use..

File: lib/ui/LandingPage.dart [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+                          onTap: () {},),
+                      style: RoundedButtonStyle.largeRoundedButtonStyle()
+                          .copyWith(buttonTextStyle: _style.actionTextStyle),
+                    ),
+                    SizedBox(height: _Constants.actionLineSpace,),
+                    GestureDetector(
+                      onTap: () => _onMenuPressed(context),
+                      child: Text(
+                        Intl.message(_viewModel.footerText),
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart]: viewmodel

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]

File: lib/ui/mockData/MockLandingPageViewModel.dart (hunk 20 above)
Comment (isaacmartinezimv) [reply lib/ui/mockData/MockLandingPageViewModel.dart]: When i retrieve the Int.message from that i forget to quit the () too, Fixed

File: lib/ui/LandingPage.dart [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+                    ),
+                  ],
+                ),
+                Column(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: <Widget>[
+                    RoundedButton(
+                      viewModel: RoundedButtonViewModel(
+                          title: Intl.message(_viewModel.actionText),
+                          onTap: () {},),
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: Action missing. Add it on the viewmodel

File: lib/ui/LandingPage.dart [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
+        ],
+      ),
+    );
+  }
+
+  Future _onMenuPressed(BuildContext context) async {
+    showModalBottomSheet(
+        context: context,
+        builder: (widgetBuilder) => ActionSheet(
+            viewModel: MockActionSheetViewModel.buildWithCheckBox(),
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart]: You need to implement your own `ActionSheetViewModel` 
Don't use the `MockActionSheetViewModel`

**Review by davidfranquet — CHANGES_REQUESTED** [review]

**Comment by davidfranquet** [comment]
@isaacmartinezimv you will need another action to handle the language change

File: lib/ui/LandingPage.dart [hunk 25]
```diff
@@ -0,0 +1,177 @@
...
+                  children: <Widget>[
+                    RoundedButton(
+                      viewModel: RoundedButtonViewModel(
+                          title: Intl.message(_viewModel.actionText),
+                          onTap: () {},),
+                      style: RoundedButtonStyle.largeRoundedButtonStyle()
+                          .copyWith(buttonTextStyle: _style.actionTextStyle),
+                    ),
+                    SizedBox(height: _Constants.actionLineSpace,),
+                    GestureDetector(
```
Comment (davidfranquet) [inline lib/ui/LandingPage.dart:181]: Give more tap area not just the text. Remember the big fingers

File: lib/ui/LandingPage.dart (hunk 23 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart]: The function now is ready, but when the ChatBot will be done, we should pass the correct function for the proper navigation

File: lib/ui/LandingPage.dart (hunk 24 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart]: Done ! Thanks for the explanation !

File: lib/ui/LandingPage.dart (hunk 25 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart:181]: Done ! Sausage finger test passed !! xD

Comment (davidfranquet) [reply lib/ui/LandingPage.dart:181]: cool

File: lib/ui/mockData/MockLandingPageViewModel.dart [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+class MockLandingPageViewModel {
+  static LandingPageViewModel build() {
+    return LandingPageViewModel(
+        detailText: Intl.message(_Strings.detailText),
+        actionText: Intl.message(_Strings.actionText),
+        footerText: Intl.message(_Strings.footerText),
+        headerImage: "assets/raw/illustration_welcome.png",
+        subtitleImage: "assets/raw/logo_default.png",
+        confirmAction: () => _mockAction(),
+      actionSheetViewModel: MockActionSheetViewModel.buildWithCheckBox()
```
Comment (davidfranquet) [inline lib/ui/mockData/MockLandingPageViewModel.dart]:
```suggestion
      actionSheetViewModel: MockActionSheetViewModel.buildWithCheckBox(),
```

and format

File: lib/ui/LandingPage.dart (hunk 23 above)
Comment (davidfranquet) [reply lib/ui/LandingPage.dart]: yes, but you can leave the function passed by the viewmodel on the onTap of the rounded button

File: lib/ui/LandingPage.dart (hunk 24 above)
Comment (davidfranquet) [reply lib/ui/LandingPage.dart]: 👍

File: lib/ui/LandingPage.dart [hunk 27]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+class LandingPageViewModel {
+  String detailText;
+  String actionText;
+  String footerText;
+
+  String headerImage;
+  String subtitleImage;
+  Function confirmAction;
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart]: what do we confirm? --- would "continueAction" be better maybe?

File: lib/ui/LandingPage.dart [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+    TextStyle detailTextStyle,
+    TextStyle footerTextStyle,
+    int detailTextMaxLines,
+    int footerTextMaxLines,
+  });
+}
+
+const LandingPageStyle _defaultStyle = const _DefaultStyle();
+
+class LandingPage extends StatelessWidget {
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:118]: good naming (no widget, no context repeating) 👍

**Review by davidfranquet — DISMISSED** [review]

File: lib/ui/LandingPage.dart (hunk 27 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart]: Perfect !

File: lib/ui/mockData/MockLandingPageViewModel.dart (hunk 26 above)
Comment (isaacmartinezimv) [reply lib/ui/mockData/MockLandingPageViewModel.dart]: normally in the mock files i didn't  check the "commas perfect style " ;)

**Review by davidfranquet — DISMISSED** [review]

**Review by LeeMatthewHiggins — DISMISSED** [review]

**Review by davidfranquet — DISMISSED** [review]

File: lib/ui/LandingPage.dart [hunk 29]
```diff
@@ -0,0 +1,224 @@
...
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static final englishAction = "English";
+  static final swahiliAction = "Swahili";
+  static final confirmAction = "Confirm";
+  static final cancelAction = "Cancel";
+  static final englishIcon = "assets/icons/flag_usa.png";
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:23]: maybe have the icons in their own _Icons class.

File: lib/ui/LandingPage.dart [hunk 30]
```diff
@@ -0,0 +1,224 @@
...
+  static final swahiliIcon = "assets/icons/flag_kenya.png";
+  static final checkBoxIcon = "assets/icons/radio_button_default.png";
+}
+
+class _Constants {
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:28]: 👍

File: lib/ui/LandingPage.dart [hunk 31]
```diff
@@ -0,0 +1,224 @@
...
+    this.headerImage,
+    this.subtitleImage,
+    this.footerText,
+    this.continueAction,
+    this.actionSheetViewModel,
+  });
+}
+
+class LandingPageStyle {
+  final TextStyle detailTextStyle;
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:62]: Nice, just the things you want people to be able to tweak 👍

File: lib/ui/LandingPage.dart [hunk 32]
```diff
@@ -0,0 +1,224 @@
...
+                ),
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  void _languageTapped(ActionSheet sheet, BuildContext context) {
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:204]: great work 👍

File: lib/ui/LandingPage.dart [hunk 33]
```diff
@@ -0,0 +1,224 @@
...
+    ActionSheet.present(sheet, context);
+  }
+
+  ActionSheet _languageMenu() {
+    final actions = [
+      ActionSheetListItemViewModel(
+        title: Intl.message(_Strings.englishAction),
+        type: ActionType.selectable,
+        icon: _Strings.englishIcon,
+        checkBoxIcon: _Strings.checkBoxIcon,
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart]: what about the functions? we should call a function that the viewModel of the landingPage provides. viewModel.changeLanguage(the language iso code)

**Review by LeeMatthewHiggins — COMMENTED** [review]
Very clean great PR.  Just need to complete the language switch feature by calling a function provided by the viewModel.

File: lib/ui/LandingPage.dart (hunk 29 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart:23]: Perfect !

File: lib/ui/LandingPage.dart (hunk 33 above)
Comment (isaacmartinezimv) [reply lib/ui/LandingPage.dart]: Done!

File: lib/ui/LandingPage.dart [hunk 34]
```diff
@@ -0,0 +1,236 @@
...
+  }
+
+  ActionSheet _languageMenu() {
+    final actions = [
+      ActionSheetListItemViewModel(
+        title: Intl.message(_Strings.englishAction),
+        type: ActionType.selectable,
+        icon: _Icons.englishIcon,
+        checkBoxIcon: _Icons.checkBoxIcon,
+        onTap: () => _viewModel.switchLanguageTapped(_Languages.english),
```
Comment (LeeMatthewHiggins) [inline lib/ui/LandingPage.dart:215]: nice,    @DavidFD005 has a PR which has the real function to switch. would be good to sync up to hook this up in another PR.

**Review by LeeMatthewHiggins — APPROVED** [review]
Great work

**Review by jessicahernandezhv — APPROVED** [review]
